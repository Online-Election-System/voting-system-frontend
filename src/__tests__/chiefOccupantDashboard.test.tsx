import React from 'react';
import { render } from '@testing-library/react';

// Mock RoleGuard to simply render children
jest.mock('@/components/auth/RoleGuard', () => {
  return ({ children }: { children: React.ReactNode }) => <>{children}</>;
});

// No-op for next/image in tests (already mocked in jest.setup but double-ensure)
jest.mock('next/image', () => (props: any) => {
  // eslint-disable-next-line @next/next/no-img-element
  return <img {...props} alt={props.alt || 'image'} />;
});

import ChiefOccupantDashboard from '../app/chief-occupant/dashboard/page';

describe('ChiefOccupantDashboard UI', () => {
  it('renders consistently (snapshot)', () => {
    const { container } = render(<ChiefOccupantDashboard />);
    expect(container).toMatchSnapshot();
  });
});
