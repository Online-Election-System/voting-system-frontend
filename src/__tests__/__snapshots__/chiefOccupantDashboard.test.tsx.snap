// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ChiefOccupantDashboard UI renders consistently (snapshot) 1`] = `
<div>
  <div
    class="flex min-h-screen flex-col"
  >
    <main
      class="flex flex-1 flex-col items-center justify-center p-4 md:p-6"
    >
      <div
        class="max-w-3xl text-center"
      >
        <h1
          class="text-4xl font-bold tracking-tight sm:text-5xl md:text-6xl"
        >
          Welcome to the Election System
        </h1>
        <p
          class="mt-4 text-lg text-muted-foreground md:text-xl"
        >
          Manage your enrollments, view upcoming elections, and stay informed about your voting status â€” all in one place.
        </p>
        <div
          class="mt-8"
        >
          <img
            alt="Election System Illustration"
            class="mx-auto"
            height="400"
            src="/election-illustration.svg?height=400&width=600"
            width="600"
          />
        </div>
      </div>
    </main>
  </div>
</div>
`;
